### ISSUE TITLE
New kind of Docker connection failure

### ISSUE THREAD

**Issue by WillEngler** [Garden-AI/garden#480]
When onboarding, Jason ran into this Docker error I haven't seen before. Given that he just installed the Garden CLI today/yesterday, it seems plausible that he ran into [this issue](https://github.com/docker/docker-py/issues/3256) that a lot of people have been running into. A possible fix is limiting `requests` to <= 2.31 until we do a deeper dive.

![image (2)](https://github.com/Garden-AI/garden/assets/6283143/d0b566e1-35ca-4006-9352-9c342f4865a8)
![image (1)](https://github.com/Garden-AI/garden/assets/6283143/c8a2ea17-ccf4-49cb-935d-12b45643e3a1)

# Acceptance Criteria
Given someone has just installed the latest release of Garden with pipx, when they try to use `notebook start`, then they can connect to Docker from the CLI.

**Comment by hholb** [comment]
@MaxTuecke Is this the same issue you ran into?

**Comment by mtilyxuegao** [comment]
I tried pinning requests 2.31 and 2.29 but it doesn't seem to help

**Comment by WillEngler** [comment]
I did a fresh install of Garden that pulled in requests 2.32, and was able to reproduce the issue. I then tweaked pyproject.toml to limit requests <=2.31. I refreshed my dependencies with `poetry update`, which installed requests 2.31. Then the issue was resolved. This would be a good first issue for a new intern to make a PR for :)

<img width="454" alt="Screenshot 2024-06-12 at 11 15 15 AM" src="https://github.com/Garden-AI/garden/assets/6283143/7c858a9b-1587-4559-9d14-905c5b4d66a5">
